pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/V2Operator.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pinot.query.runtime.operator;
+
+import java.util.List;
+import java.util.Set;
+import org.apache.pinot.core.common.Operator;
+import org.apache.pinot.core.operator.BaseOperator;
+import org.apache.pinot.query.mailbox.MailboxIdentifier;
+import org.apache.pinot.query.runtime.blocks.TransferableBlock;
+
+
+public abstract class V2Operator extends BaseOperator<TransferableBlock> implements ScheduledOperator {
+
+  @Override
+  public ScheduleResult shouldSchedule(Set<MailboxIdentifier> availableMail) {
```

Comment:
I think chaining shouldSchedule/ isBlocked API is a great solution to understand whether a OpChain is blocked or not.

However, this API should not be tied with mailbox. Same for scheduling. 
Imagine in the future, we have disk spilling for some operator to be blocked, this is not mailbox anymore. 
The way to address this is to chain event listeners together.  And use this listener to understand whether if it is blocking or not.